### PR TITLE
Feat/misc/navigation integration

### PR DESCRIPTION
**Pull request by agonzalez-r**
# feat/misc/navigation-integration : Add navigation to the screens 
## Description
This PR introduces closes issue #42 by adding `navigationActions` to all the previously added screens, in order to enable the flow between them. It also ensures the Bottom menu navigation menu and top bar are present and correctly placed in the concerned screens. 

The test were supposed to mock `navigationActions` but due to improper implementation of Mockito in the dependencies, this was not possible, therefore I call `NavigationActions(rememberNavController())` in the parameter of the Screens as the navigationActions, as a temporary fix.

#### Added
- `EditProfile.kt` and Timer.kt with placeholder texts for now 
#### Modified
- All UI screens were modified at some degree to add the navigation actions on onClick functions primarily 
- `MapScreen.kt`: heavily modified by adding a `Scaffold` to encapsulate the Map, which lead to some modularisation 
- `MainActivity.kt`: to implement PeriodPalsApp with the navigations

## Testing
- `NavigationActionsTest.kt`: tests all navigation functions 
- The tests had to be modified, all the actions on the test that called a navigation action in the Screen's function were commented 

## Screenshots
Screenshot 2024-10-18 at 05 18 25
Screenshot 2024-10-18 at 05 18 33
Screenshot 2024-10-18 at 05 18 44

### CODE REVIEW

File: app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt [hunk 1]
```diff
@@ -75,42 +77,46 @@ class SignInScreenTest {
...
+  //  @Test
+  //  fun signInScreen_navigatesToSignUp() {
+  //    composeTestRule.setContent { SignInScreen(NavigationActions(rememberNavController())) }
+  //
+  //    // Click on the "Not registered yet? Sign up here!" text
+  //    composeTestRule.onNodeWithTag("signInNotRegistered").performClick()
+  //
+  //    // Check for a navigation action (mocking would be required here)
+  //    // You would verify that the navigation to the sign-up screen is triggered.
+  //  }
```
Comment (coaguila) [inline app/src/androidTest/java/com/android/periodpals/ui/authentication/SignInTest.kt:121]: Nitpick: Perhaps remove the commented out test, we can always get it back from previous commits

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt [hunk 2]
```diff
@@ -137,6 +139,31 @@ fun SignInScreen() {
...
+                            //                              } else {
+                            //                                Toast.makeText(context, "Login
+                            // Failed", Toast.LENGTH_SHORT).show()
+                            //                              }
+                            //                            } else {
+                            //                              Toast.makeText(
+                            //                                      context, "Invalid email or
+                            // password.", Toast.LENGTH_SHORT)
+                            //                                  .show()
+                            //                            }
```
Comment (coaguila) [inline app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:166]: Nitpick: Same as before perhaps we can remove.

**Review by coaguila — COMMENTED** [review]
LGTM, there is just some nitpicks, great job 👍🏽

File: app/src/main/java/com/android/periodpals/ui/alert/AlertScreen.kt [hunk 3]
```diff
@@ -50,11 +64,11 @@ fun AlertScreen() {
                   textAlign = TextAlign.Center,
                   style = MaterialTheme.typography.titleSmall)
 
-              // Product selection
+              // Product
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertScreen.kt:67]: I like the cleaner comments

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt (hunk 2 above)
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:166]: I'm not sure what this is? Should it be uncommented as said in the todo?

File: app/src/main/java/com/android/periodpals/ui/profile/EditProfile.kt
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/profile/EditProfile.kt:1]: Can you add a comment explaining that this is a placeholder screen

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:1]: Can you add a comment explaining that this is a placeholder screen

**Review by charliemangano — APPROVED** [review]
LGTM, see comments for nitpicks

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt (hunk 2 above)
Comment (agonzalez-r) [reply app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:166]: Indeed, forgot to uncomment

Comment (francelu) [reply app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:166]: It's some duplicated code for Toasts in the `onClick`. Will remove it for the merge.

**Review by coaguila — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**